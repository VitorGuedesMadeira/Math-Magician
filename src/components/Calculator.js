/* eslint-disable react/prefer-stateless-function */
/* eslint-disable react/destructuring-assignment */
import React from 'react';
import calculate from '../logic/calculate';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: 0,
      next: null,
      operation: null,
    };
  }

  handleClick = (event) => {
    const res = calculate(this.state, event.target.textContent);
    this.setState(res);
  }

  render() {
    const { total, next } = this.state;
    return (
      <section className="calculator">
        <div className="display">{next || total}</div>
        {/* display calculation */}

        <button onClick={this.handleClick} className="btns hover" type="button">AC</button>
        <button onClick={this.handleClick} className="btns hover" type="button">+/-</button>
        <button onClick={this.handleClick} className="btns hover" type="button">%</button>
        <button onClick={this.handleClick} className="operators hover" type="button">รท</button>
        {/* first row */}

        <button onClick={this.handleClick} className="btns hover" type="button">7</button>
        <button onClick={this.handleClick} className="btns hover" type="button">8</button>
        <button onClick={this.handleClick} className="btns hover" type="button">9</button>
        <button onClick={this.handleClick} className="operators hover" type="button">x</button>
        {/* second row */}

        <button onClick={this.handleClick} className="btns hover" type="button">4</button>
        <button onClick={this.handleClick} className="btns hover" type="button">5</button>
        <button onClick={this.handleClick} className="btns hover" type="button">6</button>
        <button onClick={this.handleClick} className="operators hover" type="button">-</button>
        {/* third row */}

        <button onClick={this.handleClick} className="btns hover" type="button">1</button>
        <button onClick={this.handleClick} className="btns hover" type="button">2</button>
        <button onClick={this.handleClick} className="btns hover" type="button">3</button>
        <button onClick={this.handleClick} className="operators hover" type="button">+</button>
        {/* fourth row */}

        <button onClick={this.handleClick} className="zero hover" type="button">0</button>
        <button onClick={this.handleClick} className="btns hover" type="button">.</button>
        <button onClick={this.handleClick} className="operators hover" type="button">=</button>
        {/* fifth row */}
      </section>
    );
  }
}

export default Calculator;
