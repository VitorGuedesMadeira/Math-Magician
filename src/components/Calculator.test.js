import { fireEvent, render, screen } from '@testing-library/react';
import Calculator from './Calculator';

describe('footer component', () => {
  test('testing if the word "math!" is on it', () => {
    render(<Calculator />);
    const calculatorText = screen.getByText(/math!/);
    expect(calculatorText).toBeInTheDocument;
  });

  test('testing if theres a 2022 text in the footer', () => {
    render(<Calculator />);
    const calculatorBtn = screen.getByText(/AC/);
    expect(calculatorBtn).toBeInTheDocument;
  });

  test('testing if AC has a class', () => {
    render(<Calculator />);
    const calculatorBtn = screen.getByText(/AC/);
    expect(calculatorBtn).toHaveClass;
  });
});

describe('Calculator tests', () => {
    test('Renders Calculator component', () => {
      const tree = render(<Calculator />)
      expect(tree).toMatchSnapshot();
    });

    test('Simulating user interaction', () => {
        render(<Calculator />);
        const displayNumber = document.querySelector('.display');
        const key7 = screen.getByText('7');
        const key8 = screen.getByText('8');
        fireEvent.click(key7);
        fireEvent.click(key8);
        expect(displayNumber.textContent).toBe('78');
      });

      test('Simulating user interaction', () => {
        render(<Calculator />);
        
        const displayNumber = document.querySelector('.display');
        const key3 = screen.getByText('3');
        const key7 = screen.getByText('7');
        const keyTimes = screen.getByText('x');
        const keyEquals = screen.getByText('=');

        fireEvent.click(key3);
        fireEvent.click(keyTimes);
        fireEvent.click(key7);
        fireEvent.click(keyEquals);

        expect(displayNumber.textContent).toBe('21');
      });
});
