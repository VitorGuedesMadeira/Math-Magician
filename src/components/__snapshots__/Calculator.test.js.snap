// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator tests Renders Calculator component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="main-container"
      >
        <h2>
          Let's do some math!
        </h2>
        <div
          class="calculator"
        >
          <div
            class="display"
          >
            0
          </div>
          <button
            class="btns hover"
            type="button"
          >
            AC
          </button>
          <button
            class="btns hover"
            type="button"
          >
            +/-
          </button>
          <button
            class="btns hover"
            type="button"
          >
            %
          </button>
          <button
            class="operators hover"
            type="button"
          >
            รท
          </button>
          <button
            class="btns hover"
            type="button"
          >
            7
          </button>
          <button
            class="btns hover"
            type="button"
          >
            8
          </button>
          <button
            class="btns hover"
            type="button"
          >
            9
          </button>
          <button
            class="operators hover"
            type="button"
          >
            x
          </button>
          <button
            class="btns hover"
            type="button"
          >
            4
          </button>
          <button
            class="btns hover"
            type="button"
          >
            5
          </button>
          <button
            class="btns hover"
            type="button"
          >
            6
          </button>
          <button
            class="operators hover"
            type="button"
          >
            -
          </button>
          <button
            class="btns hover"
            type="button"
          >
            1
          </button>
          <button
            class="btns hover"
            type="button"
          >
            2
          </button>
          <button
            class="btns hover"
            type="button"
          >
            3
          </button>
          <button
            class="operators hover"
            type="button"
          >
            +
          </button>
          <button
            class="zero hover"
            type="button"
          >
            0
          </button>
          <button
            class="btns hover"
            type="button"
          >
            .
          </button>
          <button
            class="operators hover"
            type="button"
          >
            =
          </button>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="main-container"
    >
      <h2>
        Let's do some math!
      </h2>
      <div
        class="calculator"
      >
        <div
          class="display"
        >
          0
        </div>
        <button
          class="btns hover"
          type="button"
        >
          AC
        </button>
        <button
          class="btns hover"
          type="button"
        >
          +/-
        </button>
        <button
          class="btns hover"
          type="button"
        >
          %
        </button>
        <button
          class="operators hover"
          type="button"
        >
          รท
        </button>
        <button
          class="btns hover"
          type="button"
        >
          7
        </button>
        <button
          class="btns hover"
          type="button"
        >
          8
        </button>
        <button
          class="btns hover"
          type="button"
        >
          9
        </button>
        <button
          class="operators hover"
          type="button"
        >
          x
        </button>
        <button
          class="btns hover"
          type="button"
        >
          4
        </button>
        <button
          class="btns hover"
          type="button"
        >
          5
        </button>
        <button
          class="btns hover"
          type="button"
        >
          6
        </button>
        <button
          class="operators hover"
          type="button"
        >
          -
        </button>
        <button
          class="btns hover"
          type="button"
        >
          1
        </button>
        <button
          class="btns hover"
          type="button"
        >
          2
        </button>
        <button
          class="btns hover"
          type="button"
        >
          3
        </button>
        <button
          class="operators hover"
          type="button"
        >
          +
        </button>
        <button
          class="zero hover"
          type="button"
        >
          0
        </button>
        <button
          class="btns hover"
          type="button"
        >
          .
        </button>
        <button
          class="operators hover"
          type="button"
        >
          =
        </button>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
